homeassistant/components/delijn/sensor.py

```diff
@@ -37,67 +42,64 @@
 async def async_setup_platform(hass, config, async_add_entities, discovery_info=None):
     """Create the sensor."""
     api_key = config[CONF_API_KEY]
-    name = DEFAULT_NAME
 
     session = async_get_clientsession(hass)
 
     sensors = []
     for nextpassage in config[CONF_NEXT_DEPARTURE]:
-        stop_id = nextpassage[CONF_STOP_ID]
-        number_of_departures = nextpassage[CONF_NUMBER_OF_DEPARTURES]
-        line = Passages(
-            hass.loop, stop_id, number_of_departures, api_key, session, True
+        sensors.append(
+            DeLijnPublicTransportSensor(
+                Passages(
+                    hass.loop,
+                    nextpassage[CONF_STOP_ID],
+                    nextpassage[CONF_NUMBER_OF_DEPARTURES],
+                    api_key,
+                    session,
+                    True,
+                )
+            )
         )
-        await line.get_passages()
-        if line.passages is None:
-            _LOGGER.warning("No data received from De Lijn")
-            return
-        sensors.append(DeLijnPublicTransportSensor(line, name))
 
     async_add_entities(sensors, True)
 
 
 class DeLijnPublicTransportSensor(Entity):
     """Representation of a Ruter sensor."""
 
-    def __init__(self, line, name):
+    def __init__(self, line):
         """Initialize the sensor."""
         self.line = line
         self._attributes = {ATTR_ATTRIBUTION: ATTRIBUTION}
-        self._name = name
+        self._name = None
         self._state = None
-        self._available = False
 
     async def async_update(self):
         """Get the latest data from the De Lijn API."""
-        await self.line.get_passages()
-        if self.line.passages is None:
-            _LOGGER.warning("No data received from De Lijn")
-            return
         try:
+            await self.line.get_passages()
+            self._name = await self.line.get_stopname()
+        except HttpException:
+            self._state = STATE_UNAVAILABLE
+            _LOGGER.error("Delijn http error")
+
+        try:
```

Comment:
Should this be `else:`?
```suggestion
        else:
```

Reply:
Nevermind, I didn't see the except clause down below.